st2common/st2common/hooks.py

```diff
@@ -114,6 +115,12 @@ def before(self, state):
         if state.request.method == 'OPTIONS':
             return
 
+        # Token request is authenticated separately.
+        if (state.request.method == 'POST' and (
+                state.request.path == '/tokens' or
+                re.search('/v\d+/tokens$', state.request.path))):
```

Comment:
Yeah, that's kinda nasty, but I don't think we can do much else at this point. 

For one, we could maybe make path regex a constant or something, but that's about it.